datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/DropDownViewHolderFactory.kt

```diff
@@ -67,19 +68,26 @@ internal object DropDownViewHolderFactory :
           this.questionnaireViewItem.answerOption
             .map {
               DropDownAnswerOption(
+                it.value.identifierString(context),
```

Comment:
what are the cases when Type.id is null? Can you list a couple of examples? If it makes sense, then we can do the change below:

```suggestion
                it.value.let { it.id ?: it.displayString(context)},
```

Reply:
One obvious case would be when using an x-fhir-query as answerExpression for questionnaire items of type `reference`,  the resulting references in the answer options do not have their `Reference.id` populated and are null